packages/aws-cdk-lib/aws-rds/lib/cluster.ts

```diff
@@ -685,6 +685,18 @@ abstract class DatabaseClusterNew extends DatabaseClusterBase {
     };
   }
 
+  /**
+   * The ARN of the cluster
+   */
+  public get clusterArn(): string {
+    return Stack.of(this).formatArn({
+      service: 'rds',
+      resource: 'cluster',
+      arnFormat: ArnFormat.COLON_RESOURCE_NAME,
+      resourceName: this.clusterIdentifier,
+    });
+  }
+
```

Comment:
Just like `ServerlessCluster`

https://github.com/aws/aws-cdk/blob/1f30b5d873b7a1b11733211fa9ef47792fa4c678/packages/aws-cdk-lib/aws-rds/lib/serverless-cluster.ts#L345-L355

Reply:
@blimmer Is this how CDK normally handles this? I ask because the output becomes a `Join` instead of doing a `GetAtt` on the resource. Trying to wrap my head around why we should do this over the `GetAtt` path. I tried looking at the Lambda Function L2, since there is a `readonly` property for the Arn.

Honestly not sure if it matters (likely not) but trying to learn different parts of this as I go. To me seems like we have two patterns going on, the one you have here and the `readonly` property for the Arn.

Reply:
I'm not sure - I just followed the pattern in the linked code. One of the core contributors might be able to tell us.

Reply:
I asked the team directly. I did find SQS which does this differently. https://github.com/aws/aws-cdk/blob/v2.130.0/packages/aws-cdk-lib/aws-sqs/lib/queue.ts#L437-L440

```
const queue = new sqs.Queue(this, 'CdkExampleQueue', {
      visibilityTimeout: Duration.seconds(300)
    });

new CfnOutput(this, "arn", {
      exportName: "arn",
      value: queue.queueArn,
    })
```


CFN Synth
```
Outputs:
  arn:
    Value:
      Fn::GetAtt:
        - CdkExampleQueue7618E31B
        - Arn
    Export:
      Name: arn
```

Which fits my mental model better. I am not sure if there is a big difference or a reason it was done this way in RDS. Maybe one is a "legacy" way of doing things. 

Generally, this looks fine. I just want to get some deeper clarity on what is typical.